src/autoscaling/snapshotter.js

```diff
@@ -222,13 +223,33 @@ class Snapshotter {
             });
 
             this.memorySnapshots.push(snapshot);
+            this._memoryOverloadWarning(memInfo);
         } catch (err) {
             log.exception(err, 'Snapshotter: Memory snapshot failed.');
         } finally {
             intervalCallback();
         }
     }
 
+    /**
+     * Checks for critical memory overload and logs it to the console.
+     * @ignore
+     * @param {Object} memoryInfo - Memory information
+     */
+    _memoryOverloadWarning(memoryInfo) {
+        try {
+            const criticalOverloadRation = this.maxUsedMemoryRatio * CRITICAL_OVERLOAD_RATIO;
```

Comment:
Here is potential problem:

We compute `criticalOverloadRation` as `this.maxUsedMemoryRatio * CRITICAL_OVERLOAD_RATIO` where `CRITICAL_OVERLOAD_RATIO=1.2`.

Default `maxUsedMemoryRatio` is `0.7` (5.6GB out of 8GB). You can increase this limit when creating `AutoscaledPool` to let's say `0.9` which makes sense for some stable use cases. Then `criticalOverloadRation = 0.9 * 1.2 = 1,08` which will never be reached since we get OOM at `1.0`.

So I think that we should define that limit as for example `X% of remaining memory after exceeding maxUsedMemoryRatio`.

For example - let say
* `X = 0.5`
* `maxMemoryBytes=8192 MB`
* `maxUsedMemoryRatio=0.9`
then `const criticalOverloadRation = X * maxMemoryBytes * (1 - maxUsedMemoryRatio) = 0.5*8192*(1 - 0.9) = 409MB`